src/com/facebook/buck/apple/UmbrellaHeaderGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.facebook.buck.apple;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+
+import org.stringtemplate.v4.ST;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Creates an umbrella header from a list of paths
+ */
+public class UmbrellaHeaderGenerator {
+  private static final String UMBRELLA_HEADER_TEMPLATE = "umbrella_header.st";
+
+  public static String generateUmbrellaHeader(String frameworkName, Iterable<Path> headers) {
```

Comment:
I think you don't really need a separated class for this, just need a method in ProjectGenerator. should also put it inside project_generator package.
If you really want it, you should make it an utility class, see https://github.com/shatlykuber/buck/blob/f103eebf4f162efb6503780120f043e7dbcaf3fb/src/com/facebook/buck/apple/AppleDescriptions.java#L100-L100